src/clients/asset-client.js

```diff
@@ -14,42 +18,93 @@ class AssetClient {
     this.directory = directory
   }
 
+  /**
+   *
+   * @param {File} file A file object.
+   * @param {Object} options An options object allowing to specify onProgress and doAbort functions.
+   * onProgress handles upload progress in app utilizing this client. doAbort function returning true/false indicating the need to abort upload.
+   *
+   * @return {Promise} The request promise.
+   */
   create (file, options = {}) {
     return this.__createOrUpdate({
       method: 'POST',
       url: `${this.client.baseUrl}`,
       file: file,
-      onProgress: options.onProgress
+      onProgress: options.onProgress,
+      doAbort: options.doAbort
     })
   }
 
+  /**
+   *
+   * @param {String} id An id of asset to destroy.
+   * @param {Object} parameters Optional url parameters.
+   *
+   * @return {Promise} The request promise.
+   */
   destroy (id, parameters = {}) {
     return this.client.destroy(id, parameters)
   }
 
+  /**
+   *
+   * @param {Object} parameters Optional filter parameters.
+   *
+   * @return {Promise} The request promise.
+   */
   search (parameters = {}) {
     return this.client.search(parameters)
   }
 
+  /**
+   *
+   * @param {String} id An id of asset to find.
+   * @param {Object} parameters Optional url parameters.
+   *
+   * @return {Promise} The request promise.
+   */
   find (id, parameters = {}) {
     return this.client.find(id, parameters)
   }
 
+  /**
+   *
+   * @param {File} file A file object.
+   * @param {String} id An id of asset to update.
+   * @param {Object} options An options object allowing to specify onProgress and doAbort functions.
+   * onProgress handles upload progress in app utilizing this client. doAbort function returning true/false indicating the need to abort upload.
+   *
+   * @return {Promise} The request promise.
+   */
   update (file, id, options = {}) {
     return this.__createOrUpdate({
       method: 'PUT',
       url: `${this.client.baseUrl}/${id}`,
       file: file,
-      onProgress: options.onProgress
+      onProgress: options.onProgress,
+      doAbort: options.doAbort
     })
   }
 
+  /**
+   *
+   * @param {String} id An id of asset to get the temporary download url of.
+   *
+   * @return {URL} The asset temporary download url.
```

Comment:
Actually not true, this function still returns a promise. We should write something like
`@return {Promise} A promise that resolves to a temporary asset download url.`